services/Referral.js

```diff
@@ -19,6 +22,84 @@ class Referral {
     }
   }
 
+  /**
+   * @description Creates a referral link in shortened URL form
+   * @param {string} email
+   * @param {string} name
+   * @returns {object} response
+   */
+  static async createReferralLink(email, name) {
+    let longFormEncodedUrl;
+    let referralLink;
+
+    // Check first if link already exists in our DB
+    try {
+      referralLink = await ReferralLink.findOne({
+        email
+      });
+
+      if (referralLink) {
+        Logger.info(`Link already created for ${email}, returning entry in DB`);
+        return referralLink;
+      }
+    } catch (e) {
+      Logger.error(`Error finding referral link for ${email}`, e);
+    }
+
+    // Create long form encoded URL
+    try {
+      longFormEncodedUrl = constructEncodedSALReferralUrl(name, email);
```

Comment:
What does "SAL" mean in `constructEncodedSALReferralUlr`

Reply:
`Support A Learner`